### DISCUSSION TITLE
target a specific dependency in a pom.xml

### DISCUSSION THREAD

**Discussion by Ashley-Emery** [renovatebot/renovate#25577] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

36.23.0

### If you're self-hosting Renovate, select which platform you are using.

Bitbucket Cloud (bitbucket.org)

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

For about a year now we've been running renovate and so far it's been great. Currently I'm having trouble with the Maven manager. Wondering if there's a way to target a specific dependency in a pom.xml.

What's implemented so far works but we're getting PRs for each dependency in it and I'm only interested in a specific one.

```
<dependency>
    <groupId>org.camunda.bpm</groupId>
    <artifactId>camunda-bom</artifactId>
    <version>7.19.5-ee</version>
    <scope>import</scope>
    <type>pom</type>
</dependency>
```

I would appreciate any ideas.

Thanks in advance!

### What you tried so far.

```
{
  "enabled": true,
  "enabledManagers": ["maven"],
  "logFormat": "json",
  "baseBranches": ["master"],
  "packageRules": [
    {
      "matchPackageNames": ["org.camunda.bpm:camunda-bom"],
      "versioning": "maven",
      "versionCompatibility": "^(?<version>[^-]+)(?<compatibility>-.*)?$",
      "matchManagers": ["maven"],
      "matchDatasources": ["maven"],
    },
  ],
  "repositories": [
      {
          "repository": 'xxxxxxxx/camunda-service'
      },
  ]
}
```

**Comment by rarkins** [comment]
Try this package rule:

```json
{
  "matchManagers": ["maven"],
  "excludePackageNames": ["org.camunda.bpm:camunda-bom"],
  "enabled": false
}
```

i.e. disable all maven dependencies except org.camunda.bpm:camunda-bom

**Comment by Ashley-Emery** [comment]
Renovate keeps creating PRs for all the dependencies in the pom.xml. The reason I'm trying to target just an specific dependency is because we have multiple repositories on-boarded with renovate that also have a pom.xml. Renovate's scanning all these other repositories and we don't want/need the maven manager doing that.

Would the `matchRepositories` be useful in this case? Since we only want to target the svc-camunda repository. Or a way to target the maven manager to just one specific repository...

I just need the maven manager to work with a single repository and to create a PR for `org.camunda.bpm:camunda-bom` and ignore the rest of the dependencies in the file.

Something like this? 
```
{
  "enabled": true,
  "enabledManagers": ["maven"],
  "logFormat": "json",
  "baseBranches": ["master"],
  "branchConcurrentLimit": 5,
  "timezone": "America/Knox_IN",
  "schedule": ["after 8am every weekday", "before 5pm every weekday"],
  "maven": {
    "fileMatch": [
      "(^|/|\\.)pom\\.xml$",
    ],
  },
  "packageRules": [
    {
      "description": "upgrade svc-camunda bom dependency",
      "matchManagers": ["maven"],
      "matchDatasources": ["maven"],
      "matchPackageNames": ["org.camunda.bpm:camunda-bom"],
      "matchRepositories": ["XXXXXXXX/svc-camunda"],
      "versioning": "maven",
      "versionCompatibility": "^(?<version>[^-]+)(?<compatibility>-.*)?$",
    },
  ],
  "repositories": [
      {
          "repository": 'XXXXXXXX/svc-camunda'
      },
      {
          "repository": 'XXXXXXXX/portal-accounts'
      },
      {
          "repository": 'XXXXXXXX/portal-applications'
      },
      {
          "repository": 'XXXXXXXX/portal-apv'
      },
      {
          "repository": 'XXXXXXXX/portal-education'
      },
      {
          "repository": 'XXXXXXXX/portal-groups'
      },
      {
          "repository": 'XXXXXXXX/portal-payments'
      },
      {
          "repository": 'XXXXXXXX/portal-policy'
      },
      {
          "repository": 'XXXXXXXX/portal-quote'
      },
      },
  ]
}
```

Thanks in advance!

**Reply by rarkins** [reply]
I've already given you a recommended config
